api/server/router/image/image_routes.go

```diff
@@ -67,10 +69,39 @@ func (ir *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrit
 			}
 		}
 
+		// Special case: "pull -a" may send an image name with a
+		// trailing :. This is ugly, but let's not break API
+		// compatibility.
+		image := strings.TrimSuffix(img, ":")
```

Comment:
LOL, and I went looking for the history behind this one, and what a great commit I found (looks like that one did the exact reverse as this PR is doing :joy: https://github.com/moby/moby/commit/47afe6bd0a0dc806c36d0978a8d3e23f75ef87fe)


In either case; this comment was originally added in https://github.com/moby/moby/commit/4352da7803d182a6013a5238ce20a7c749db979a (was hoping there was more context to it, but din't see so at a glance)